modules/@apostrophecms/schema/ui/apos/logic/AposInputSlug.js

```diff
@@ -169,39 +179,23 @@ export default {
         preserveDash = true;
       }
 
-      s = sluggo(s, options);
+      let slug = sluggo(s, options);
       if (preserveDash) {
-        s += '-';
+        slug += '-';
       }
 
       if (this.field.page && !componentOnly) {
-        if (!this.followingValues?.title) {
-          const nextParts = this.next.split('/');
-          if (s === nextParts[nextParts.length - 1]) {
-            s = '';
-            if (this.originalSlugPartsLength === nextParts.length) {
-              nextParts.pop();
-            }
-            this.next = nextParts.join('/');
-          }
-        }
```

Comment:
really unclear what was doing this code when title is empty (removing last part of slug?). Removed without noticing issues.